Source/Extensions/Blazorise.DataGrid/DataGridColumn.cs

```diff
@@ -35,19 +36,62 @@ public partial class DataGridColumn<TItem> : BaseDataGridColumn<TItem>
 
     public DataGridColumn()
     {
-        // TODO: move this to cached FunctionCompiler so it doesn't get compiled every time
-        valueTypeGetter = new( () => FunctionCompiler.CreateValueTypeGetter<TItem>( Field ) );
-        defaultValueByType = new( () => FunctionCompiler.CreateDefaultValueByType<TItem>( Field ) );
-        valueGetter = new( () => FunctionCompiler.CreateValueGetter<TItem>( Field ) );
-        valueSetter = new( () => FunctionCompiler.CreateValueSetter<TItem>( Field ) );
-        sortFieldGetter = new( () => FunctionCompiler.CreateValueGetter<TItem>( SortField ) );
-    }
 
+        if ( typeof( TItem ) == typeof( ExpandoObject ) )

```

Comment:
I think it would be worth it to have a non-static `FunctionCompiler` class that could be abstracted and then have the implementation for each TItem object type. Instead of having `if`s like this for each type.

Reply:
I'm not sure I follow, independent of whatever abstraction you always need to check for type of ExpandoObject.

Reply:
friendly reminder

Reply:
Meh, leave it. If we ever have more TItem types, we will optimize it.